test/parallel/test-assert.js

```diff
@@ -330,10 +331,13 @@ assert.throws(function() {assert.ifError(new Error('test error'));});
 assert.doesNotThrow(function() {assert.ifError(null);});
 assert.doesNotThrow(function() {assert.ifError();});
 
-assert.throws(() => {
-  assert.doesNotThrow(makeBlock(thrower, Error), 'user message');
-}, /Got unwanted exception. user message/,
-   'a.doesNotThrow ignores user message');
+assert.throws(
+  () => {
```

Comment:
@jasnell There are certainly other options, like assigning the function to a variable so you can just do `assert.throws(foo,...)`.